cli/internal/taskhash/taskhash.go

```diff
@@ -224,7 +224,7 @@ type taskHashInputs struct {
 	hashOfFiles          string
 	externalDepsHash     string
 	task                 string
-	outputs              []string
+	outputs              fs.TaskOutputs
```

Comment:
This _might_ be the right thing to do, but this will invalidate all caches and we'll need to bump the global cache key.

Another thing we could do instead is add a method to `TaskOutputs` like `GetHashableArray` and return a `[]string` of `Inclusions` followed by `Exclusions` w/ the `!` prepended. Since `!` doesn't work right now, I think that will be strictly additive.

Reply:
That makes sense, although then we'd be effectively converting from with `!` in the `turbo.json`, to without `!` and back to with `!` for hashing. Is that too much churn?

Reply:
Hmm. I guess the object is probably fine, since we expect to expand the use of this feature